lib/srv/discovery/fetchers/aws-sync/s3.go

```diff
@@ -224,28 +231,29 @@ func (a *awsFetcher) getS3BucketDetails(ctx context.Context, bucket *s3.Bucket,
 				failedTags:         true,
 			}, errs
 	}
+	s3Client := a.awsClients.getS3Client(awsCfg)
 
-	details.policy, err = s3Client.GetBucketPolicyWithContext(ctx, &s3.GetBucketPolicyInput{
+	details.policy, err = s3Client.GetBucketPolicy(ctx, &s3.GetBucketPolicyInput{
 		Bucket: bucket.Name,
 	})
-	if err != nil && !isS3BucketPolicyNotFound(err) {
+	if err != nil && !isNotFound(err) {
 		errs = append(errs,
 			trace.Wrap(err, "failed to fetch bucket %q inline policy", aws.ToString(bucket.Name)),
 		)
 		failedReqs.policyFailed = true
 	}
 
-	details.policyStatus, err = s3Client.GetBucketPolicyStatusWithContext(ctx, &s3.GetBucketPolicyStatusInput{
+	details.policyStatus, err = s3Client.GetBucketPolicyStatus(ctx, &s3.GetBucketPolicyStatusInput{
 		Bucket: bucket.Name,
 	})
-	if err != nil && !isS3BucketPolicyNotFound(err) {
+	if err != nil && !isNotFound(err) {
```

Comment:
Do you know if there is any reason that we were picking certain error codes before? AFAICT, the API can also return not found if the bucket name is not found.